### ISSUE TITLE
[question]: Type 'number' is not assignable to type 'RangeTuple'

### ISSUE THREAD

**Issue by kittaakos** [krisk/Fuse#368]
I would like to ask for some clarification on the `RangeTuple` when used in a `Fuse.FuseResult`.

I am total rooky with `fuse.js`, and I use it with TS.

I have this dummy module:
```ts
import Fuse from 'fuse.js';

interface Board {
    readonly name: string;
};

const list: Array<Board> = [{
    name: 'Arduino Duemilanove or Diecimila'
}];

const options: Fuse.IFuseOptions<Board> = {
    useExtendedSearch: true,
    includeMatches: true,
    minMatchCharLength: 1,
    keys: [
        'name'
    ]
};

const result = new Fuse(list, options).search(`ar due di`);
console.log(JSON.stringify(result, null, 2));
```

After the search, I print the results to the console. I have one match, it looks like this:
```json
[
  {
    "item": {
      "name": "Arduino Duemilanove or Diecimila"
    },
    "refIndex": 0,
    "matches": [
      {
        "indices": [
          [
            2,
            2
          ],
          [
            4,
            4
          ],
          [
            8,
            8
          ],
          [
            12,
            12
          ],
          [
            23,
            24
          ]
        ],
        "value": "Arduino Duemilanove or Diecimila",
        "key": "name"
      }
    ]
  }
]
```
(To be honest, I do not understand the `indices` here; why does it start with `2` instead of the first character, but that's a different topic.)

If I copy-paste the JSON from the console, and I do the following, I have a compiler error:
```ts
const oneResult: Fuse.FuseResult<Board> = {
    "item": {
        "name": "Arduino Duemilanove or Diecimila"
    },
    "refIndex": 0,
    "matches": [
        {
            "indices": [
                [
                    2,
                    2
                ],
                [
                    4,
                    4
                ],
                [
                    8,
                    8
                ],
                [
                    12,
                    12
                ],
                [
                    23,
                    24
                ]
            ],
            "value": "Arduino Duemilanove or Diecimila",
            "key": "name"
        }
    ]
}
```

```
Type 'number' is not assignable to type 'RangeTuple'.ts(2322)
```

I can manually correct the `RangeTuples` and then TS is happy:
```ts
const oneResult: Fuse.FuseResult<Board> = {
    "item": {
        "name": "Arduino Duemilanove or Diecimila"
    },
    "refIndex": 0,
    "matches": [
        {
            "indices": [
                [
                    [2, 2]
                ],
                [
                    [4, 4]
                ],
                [
                    [8, 8]
                ],
                [
                    [12, 12]
                ],
                [
                    [23, 24]
                ]
            ],
            "value": "Arduino Duemilanove or Diecimila",
            "key": "name"
        }
    ]
}
```
Is this correct?
https://github.com/krisk/Fuse/blob/81a1e9a8e4df8ade9511fe1fec01cc652fa69964/src/index.d.ts#L205

Why not `indices: ReadonlyArray<RangeTuple>`? 😊

If I do:
```ts
// See `search` setup from the above example 👆
const result = new Fuse(list, options).search(`ar due di`);
for (const { matches } of result) {
    if (!!matches) {
        for (const match of matches) {
            for (const tuples of match.indices) {
                for (const [start, end] of tuples) {
                    console.log('start', start, 'end', end);
                }
            }
        }
    }
}
```

It prints:
```
start undefined end undefined
start undefined end undefined
start undefined end undefined
start undefined end undefined
start undefined end undefined
start undefined end undefined
start undefined end undefined
start undefined end undefined
start undefined end undefined
start undefined end undefined
```

Thank you!

**Comment by kittaakos** [comment]
Maybe it is TS specific, so here are a few more details, and I use `fuse.js@5.0.9-beta`.

`tsconfig.json`:
```json
{
  "compilerOptions": {
    "target": "ES5",
    "module": "commonjs",
    "lib": [
      "ES6",
      "DOM"
    ],
    "outDir": "lib",
    "strict": true,
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "noImplicitThis": true,
    "noUnusedLocals": true,
    "noImplicitReturns": true,
    "esModuleInterop": true,
  },
  "include": [
    "src"
  ]
}
```

`package.json`:
```json
{
  "name": "fuse.js-test",
  "private": true,
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "prepare": "yarn build",
    "build": "tsc",
    "watch": "tsc -w",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "fuse.js": "^5.0.9-beta"
  },
  "devDependencies": {
    "typescript": "^3.5.1"
  }
}

```

**Comment by krisk** [comment]
The problem is that I made a mistake, and you are correct. I will fix it 😄

**Comment by krisk** [comment]
Fixed in v5.0.10-beta

**Comment by kittaakos** [comment]
> Fixed in v5.0.10-beta

Verified; the TS API works as expected. Thanks for the great support 👍

> Separately, when I try your exact case, I get the following (correct) indices.
> 
> ```ts
> [
>  [ 0, 4 ],   [ 7, 10 ],
>  [ 12, 12 ], [ 14, 14 ],
>  [ 18, 19 ], [ 21, 25 ],
>  [ 27, 27 ], [ 29, 29 ],
>  [ 31, 31 ]
> ]
> ```

How did you get different indices than I do? Please explain this to me as I am a bit puzzled with the indices I get. 👇Thank you!

> (To be honest, I do not understand the `indices` here; why does it start with `2` instead of the first character, but that's a different topic.)
